dapps/profile.vue/src/lib/profileMigration.ts

```diff
@@ -69,32 +69,66 @@ export default class ProfileMigrationLibrary {
    *
    * @param      {any}  runtime  The runtime
    */
-  static async migrateProfile(runtime) {
+  static async migrateProfile(runtime, type: string) {
     const profileAddress = runtime.profile.profileContract.options.address;
     const currentAccount = runtime.activeAccount;
 
+    const oldDataFields = ['addressBook', 'bookmarkedDapps', 'contracts', 'contacts', 'profileOptions', 'publicKey', 'templates'];
+    const newDataFields = [ 'accountDetails', 'contact', 'deviceDetails', 'registration', ];
+
     // get the profile factory
     const factoryAddress = await runtime.nameResolver.getAddress('profile.factory.evan');
     const factory = runtime.contractLoader.loadContract(
-      'BaseContractFactoryInterface',
+      'ProfileDataContractFactoryInterface',
       factoryAddress
     );
+
+    const description = {
+      public: {
+        author: '',
+        dataSchema: {
+          ...runtime.profile.accountTypes.user.template.properties,
+          ...runtime.profile.accountTypes[type].template.properties,
+        },
+        dbcpVersion: 2,
+        description: 'Container for Digital Twin Data',
+        name: 'Container Contract (DataContract)',
+        version: '0.1.0',
+      },
+    };
+
+    // format data schema to remove properties
+    Object.keys(description.public.dataSchema)
+      .forEach(property => description.public.dataSchema[property] = description.public.dataSchema[property].dataSchema);
+
+    // calculate correct dbcp description and setup empty profile data properties
+    const profileTypeObject = { accountDetails: { profileType: type },};
```

Comment:
at least for one liners I would skip the trailing comma ;) 

```suggestion
    const profileTypeObject = { accountDetails: { profileType: type }};
```